belgif-rest-problem-it/belgif-rest-problem-spring-boot-2-it/src/main/java/io/github/belgif/rest/problem/OpenApiValidationConfig.java

```diff
@@ -0,0 +1,44 @@
+package io.github.belgif.rest.problem;
+
+import java.util.Collections;
+
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import com.atlassian.oai.validator.OpenApiInteractionValidator;
+import com.atlassian.oai.validator.report.LevelResolver;
+import com.atlassian.oai.validator.report.ValidationReport;
+import com.atlassian.oai.validator.springmvc.OpenApiValidationFilter;
+import com.atlassian.oai.validator.springmvc.OpenApiValidationInterceptor;
+
+@Configuration
+public class OpenApiValidationConfig implements WebMvcConfigurer {
+
+    @Bean
+    public FilterRegistrationBean<OpenApiValidationFilter> validationFilter() {
+        FilterRegistrationBean<OpenApiValidationFilter> filterRegistration = new FilterRegistrationBean<>();
+        filterRegistration.setFilter(new OpenApiValidationFilter(true, false));
+        filterRegistration.setUrlPatterns(Collections.singletonList("/openapi-validation/*"));
+        return filterRegistration;
+    }
+
+    @Override
+    public void addInterceptors(@NonNull InterceptorRegistry registry) {
+        OpenApiInteractionValidator validator = OpenApiInteractionValidator
+                .createForSpecificationUrl("/openapi.yaml")
+                .withLevelResolver(LevelResolver.create()
+                        // Accept additionalProperties even if they're not defined in the schema
+                        .withLevel("validation.schema.additionalProperties", ValidationReport.Level.IGNORE)
+                        // Ignores validation when a path is not in the openapi and let Spring handle the error
```

Comment:
This also means this code is not used:

https://github.com/belgif/rest-problem-java/blob/da8daa2ad77026bd8235d3751e061835cabd6997/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/OpenApiRequestViolationProblemAdvice.java#L59-L61

As can be seen in the [code coverage](https://sonarcloud.io/component_measures?metric=new_coverage&selected=belgif_rest-problem-java%3Abelgif-rest-problem-spring%2Fsrc%2Fmain%2Fjava%2Fio%2Fgithub%2Fbelgif%2Frest%2Fproblem%2Fspring%2FOpenApiRequestViolationProblemAdvice.java&view=list&pullRequest=118&id=belgif_rest-problem-java).

I tested what happens if we remove this level customization, and indeed, the InvalidRequestException does NOT get sent to the `@ExceptionHandler(value = InvalidRequestException.class)`.

See also https://github.com/spring-projects/spring-framework/issues/22619#issuecomment-658661059.

The InvalidRequestException then gets converted to an HTTP 404 with a payload that is not problem-compliant.

So, if we can't figure out how to handle this case, then we should document that it is required to disable this `validation.request.path.missing` validation.

Reply:
This is only the case for SB2. 
I missed the fact that validation.request.path.missing was also disabled for sb3. (enabled it again)
That was why the test for the missing path is only disabled for SB2.

Was maybe a bit better described in the original codecomment: https://github.com/belgif/rest-problem-java/pull/118/files/3657fa3c5a92430a01110f97771df4c256c24f9f#diff-6d0d73833ab5b51decf31ceea63039d8ad16eb13250383001806d1dd40dc3dbe